blackjax/smc/tuning/from_particles.py

```diff
@@ -0,0 +1,50 @@
+"""
+strategies to tune the parameters of mcmc kernels
+used within SMC, based on particles.
+"""
+import jax
+import jax.numpy as jnp
+
+from blackjax.types import Array
+
+__all__ = [
+    "particles_means",
+    "particles_stds",
+    "particles_covariance_matrix",
+    "mass_matrix_from_particles",
+]
+
+
+def particles_stds(particles):
+    return jax.numpy.std(particles_as_rows(particles), axis=0)
+
+
+def particles_means(particles):
+    return jax.numpy.mean(particles_as_rows(particles), axis=0)
+
+
+def particles_covariance_matrix(particles):
+    return jax.numpy.cov(particles_as_rows(particles), ddof=0, rowvar=False)
+
+
+def mass_matrix_from_particles(particles) -> Array:
+    """
+    Implements tuning from section 3.1 from https://arxiv.org/pdf/1808.07730.pdf
+    Computing a mass matrix to be used in HMC from particles.
+    Given the particles covariance matrix, set all non-diagonal elements as zero,
+     take the inverse, and keep the diagonal.
+    Returns
+    -------
+    A mass Matrix
+    """
+    stds = jax.numpy.std(particles_as_rows(particles), axis=0)
+    return jnp.diag(jnp.atleast_1d(jnp.reciprocal(jnp.square(stds))))
+
+
+def particles_as_rows(particles):
```

Comment:
This would not work if you have a pytree with multi dimensional leaf. Try:
```
from jax.flatten_util import ravel_pytree
...
def particles_as_rows(particles):
    return jax.vmap(lambda x: ravel_pytree(x)[0])(particles)
```

Reply:
That test case only have dimension 2 and 1 (which then expand to 2 by your function), but if there is a 3d vector i think it will break the function below